### PR TITLE
Add coveralls.io support.

### PR DESCRIPTION
**Pull request by DentonGentry**
We include -bench because some parts of the codebase, like
smallzstd, do not have regular unit tests but do have very
good benchmark tests that covers all functions.

Signed-off-by: Denton Gentry <dgentry@tailscale.com>

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v1
+
+    - name: Basic build
+      run: go build ./cmd/...
+
+    - name: Run tests on linux with coverage data
+      run: go test -race -covermode=atomic -coverprofile=coverage.txt -bench=regex ./...
```
Comment (josharian) [inline .github/workflows/coverage.yml:32]: I think you want `bench=.` and `-benchtime=1x`.

Do we need coverage counts? If not, we can omit covermode, I believe.

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    - name: Set up Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.15
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v1
+
```
Comment (josharian) [inline .github/workflows/coverage.yml:27]: Might want to cache the Go module cache here. (Look at other GitHub workflows for how to do this; grep for `**/go.sum`.)

**Review by josharian — APPROVED** [review]

**Review by josharian — APPROVED** [review]

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (DentonGentry) [reply .github/workflows/coverage.yml]: Done.

Now I understand what -bench=regex means, it is supposed to be the regex not putting it into a regular expression mode of some kind.

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (DentonGentry) [reply .github/workflows/coverage.yml:27]: Done.

**Pull request merged**